protokube/pkg/gossip/dns/hosts.go

```diff
@@ -38,20 +38,29 @@ func (h *HostsFile) Update(snapshot *DNSViewSnapshot) error {
 			klog.Warningf("ignoring unexpected lines in /etc/hosts: %v", badLines)
 		}
 
+		hostsToAddr := make(map[string][]string)
```

Comment:
if you make a map then one hostname can only have one IP family, if that is your preference I suggest you to default always to the same one, usually ipv4. Otherwise you can create one mapping per ip familiy, this will give you a cleaner separation in the code between families

Reply:
So this is a map from hosts to ip addresses, the addresses can be a mix of ipv4 and ipv6 (i.e. we build this map by appending to the existing map entries).

I actually introduced this map because _previously_ the code did replace the addresses, so ipv4 would overwrite the ipv6 address.

Reply:
(It's confusing because this map is "backwards" compared to /etc/hosts; I find the mapping in /etc/hosts unintuitive considering how we are using it, so we try to keep the address -> names mapping isolated in `hosts.HostMap`)

Reply:
osh, so `record.Name` is an IP address?

Reply:
are not `Rrdatas`the ip addresses?

Reply:
oh, nevermind, I got it now